File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/calltree/action/ToggleShowPackageNameButton.java
package eu.nyerel.panda.ijplugin.runner.calltree.action;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.util.IconLoader;
import com.intellij.ui.ToggleActionButton;
import com.intellij.ui.treeStructure.treetable.TreeTable;
import eu.nyerel.panda.ijplugin.runner.PandaSettings;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public class ToggleShowPackageNameButton extends ToggleActionButton {

    public ToggleShowPackageNameButton() {
        super("Show Package Name", IconLoader.findIcon("/nodes/package.png"));
    }

    @Override
    public boolean isSelected(AnActionEvent e) {
        return PandaSettings.INSTANCE.isShowPackageName();
    }

    @Override
    public void setSelected(AnActionEvent e, boolean state) {
        PandaSettings.INSTANCE.setShowPackageName(state);
        DrawCallTreeAction.INSTANCE.drawInBackground(e.getProject(),
                (TreeTable) getContextComponent());
    }
}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/calltree/action/ToggleShowPackageNameButton.kt
package eu.nyerel.panda.ijplugin.runner.calltree.action

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.util.IconLoader
import com.intellij.ui.ToggleActionButton
import eu.nyerel.panda.ijplugin.runner.PandaSettings
import eu.nyerel.panda.ijplugin.runner.calltree.CallTreeDrawer

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
class ToggleShowPackageNameButton : ToggleActionButton("Show Package Name", IconLoader.findIcon("/nodes/package.png")) {

    override fun isSelected(e: AnActionEvent): Boolean {
        return PandaSettings.isShowPackageName
    }

    override fun setSelected(e: AnActionEvent, state: Boolean) {
        PandaSettings.isShowPackageName = state
        CallTreeDrawer.drawInBackground(e.project)
    }
}
